lib/game_teams_collection.rb

```diff
@@ -1,17 +1,179 @@
 require 'csv'
-require_relative 'game_team'
+require_relative 'game_team' #what's the need for inheritance when you can
+#access methods through require (instructor/mod2 support)
 require_relative 'csvloadable'
+require_relative 'games_collection'
+require_relative 'create_objects'
 
-class GameTeamsCollection
+
+class GameTeamsCollection #< StatTracker
   include CsvLoadable
+  include CreateObjects
 
   attr_reader :game_teams
 
   def initialize(game_teams_path)
     @game_teams = create_game_teams(game_teams_path)
+    all_games_by_team_id
   end
 
   def create_game_teams(game_teams_path)
     create_instances(game_teams_path, GameTeam)
   end
+
+  def highest_scoring_visitor
+    team_id_to_goal = game_teams.reduce({}) do |acc, gameteam|
+      if gameteam.hoa == "away"
+        if acc[gameteam.team_id] == nil
+          acc[gameteam.team_id] = []
+          acc[gameteam.team_id] << gameteam.goals
+        else
+          acc[gameteam.team_id] << gameteam.goals
+        end
+      end
+      acc
+    end
+    id_to_average = team_id_to_goal.reduce({}) do |acc, keyvalue|
+      id = keyvalue[0]
+      avg = (keyvalue[1].sum) / (keyvalue[1].length).to_f
+
+      acc[id] = [avg]
+      acc
+    end
```

Comment:
This is a good opportunity for a refactor. The nested if statements need to be adjusted.